access/ms-teams/config.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"github.com/gravitational/teleport-plugins/access/config"
+	"github.com/gravitational/teleport-plugins/access/ms-teams/msapi"
+	"github.com/gravitational/teleport-plugins/lib"
+	"github.com/gravitational/teleport-plugins/lib/logger"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+	"github.com/pelletier/go-toml"
+)
+
+// Config represents plugin configuration
+type Config struct {
+	Teleport   lib.TeleportConfig
+	Recipients config.RecipientsMap `toml:"role_to_recipients"`
+	Log        logger.Config
+	MSAPI      msapi.Config `toml:"msapi"`
+	Preload    bool         `toml:"preload"`
+}
+
+// LoadConfig reads the config file, initializes a new Config struct object, and returns it.
+// Optionally returns an error if the file is not readable, or if file format is invalid.
+func LoadConfig(filepath string) (*Config, error) {
+	t, err := toml.LoadFile(filepath)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	conf := &Config{}
+	if err := t.Unmarshal(conf); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	err = conf.CheckAndSetDefaults()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return conf, nil
+}
+
+// CheckAndSetDefaults checks the config struct for any logical errors, and sets default values
+// if some values are missing.
+// If critical values are missing and we can't set defaults for them — this will return an error.
+func (c *Config) CheckAndSetDefaults() error {
+	if err := c.Teleport.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if c.MSAPI.Region == "" {
+		c.MSAPI.Region = "uk"
```

Comment:
Nit: Should probably default to US (unless I'm missing what "uk" means).

Reply:
That's not clear here. MS does not have proper documentation on the region codes and bot framework REST API usage. I am not even sure that this region setting affects the real region. I've changed it to `emea` as it seems the default region which is used by the .NET MS SDKs themselves.